homeassistant/components/escea/config_flow.py

```diff
@@ -10,28 +11,39 @@
 from homeassistant.helpers import config_entry_flow
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 
-from .const import DISPATCH_CONTROLLER_DISCOVERED, ESCEA, TIMEOUT_DISCOVERY
+from .const import (
+    DISPATCH_CONTROLLER_DISCOVERED,
+    ESCEA,
+    ESCEA_FIREPLACE,
+    TIMEOUT_DISCOVERY,
+)
 from .discovery import async_start_discovery_service, async_stop_discovery_service
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def _async_has_devices(hass: HomeAssistant):
+async def _async_has_devices(hass: HomeAssistant) -> bool:
 
     controller_ready = asyncio.Event()
+    remove_handler: Callable[[], None]
 
     @callback
     def dispatch_discovered(_):
         controller_ready.set()
 
-    async_dispatcher_connect(hass, DISPATCH_CONTROLLER_DISCOVERED, dispatch_discovered)
+    remove_handler = async_dispatcher_connect(
+        hass, DISPATCH_CONTROLLER_DISCOVERED, dispatch_discovered
+    )
 
     disco = await async_start_discovery_service(hass)
 
     with suppress(asyncio.TimeoutError):
         async with timeout(TIMEOUT_DISCOVERY):
             await controller_ready.wait()
 
+    if remove_handler is not None:
```

Comment:
We don't need this check. It will always be true.

Reply:
FIxed, thank you